tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+extern bool *traced_syscalls;
+extern char *traced_syscalls_str;
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   const char *str1 = "test string";
+   const char *str1_equivalent = "test string*";
+   const char *qmark_str1 = "tes? string";
+   const char *wrong_str1 = "test string1";
+   const char *asterisk_str1 = "test *";
+   const char *str2 = "test strin1";
+   const char *invalid_str = "te*t ";
+
+   ASSERT_EQ(true, simple_wildcard_match(str1, str1));
+   ASSERT_EQ(true, simple_wildcard_match(str1, str1_equivalent));
+   ASSERT_EQ(true, simple_wildcard_match(str1, qmark_str1));
+   ASSERT_EQ(true, simple_wildcard_match(str1, asterisk_str1));
+   ASSERT_EQ(false, simple_wildcard_match(str1, str2));
+   ASSERT_EQ(false, simple_wildcard_match(str1, invalid_str));
+   ASSERT_EQ(false, simple_wildcard_match(str1, wrong_str1));
+}
+
+TEST(set_traced_syscalls_int, basic)
+{
+   tracing_mock mock;
+   const char *too_long = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+   "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+   "aaaaaa";
+   const char *almost_too_long = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+   "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+   const char *str1 = ",,,";
+   const char *normal_name = "abc";
+   const char *normal_name_commas = "a, b, c";
+   ASSERT_EQ(set_traced_syscalls_int(too_long), -ENAMETOOLONG);
+   ASSERT_EQ(set_traced_syscalls_int(almost_too_long), -ENAMETOOLONG);
+
+   EXPECT_CALL(mock, handle_sys_trace_arg(_))
+      .WillOnce(Return(1));
+   ASSERT_EQ(set_traced_syscalls_int(str1), 1);
+
+   EXPECT_CALL(mock, handle_sys_trace_arg(_))
+      .WillOnce(Return(1));
+   ASSERT_EQ(set_traced_syscalls_int(normal_name), 1);
+
+   traced_syscalls = (bool *) malloc(sizeof(bool) * MAX_SYSCALLS); // TODO clean up memory
```

Comment:
Nope.
1. It's safe to assume that sizeof(bool) == 1
2. Don't use the heap, unless there's no other choice. You can make that point to a stack array on the stack and set to NULL at the end of this function.

Reply:
Fixed